tests/phpunit/tests/post/wpUniquePostSlug.php

```diff
@@ -380,4 +380,90 @@ public function test_embed_slug_should_be_suffixed_for_attachments() {
 		$found = wp_unique_post_slug( 'embed', $p, 'publish', 'attachment', 0 );
 		$this->assertSame( 'embed-2', $found );
 	}
+
+	/**
+	 * Data provider for testing unique slug constraints.
+	 *
+	 * @return array[] Test data.
+	 */
+	public static function unique_slug_test_cases() {
```

Comment:
Generally WP gives data providers the same name as the test, replacing the prefix with `data_`.

It would also be good to move the data provider below the test as that's where they are most commonly placed.

```suggestion
	public static function data_unique_slugs() {
```